payloads/v8/channel.ts

```diff
@@ -930,3 +934,86 @@ export interface APIAllowedMentions {
 	 */
 	replied_user?: boolean;
 }
+
+/**
+ * TODO: Include docs link
+ */
+export interface APIBaseComponent {
```

Comment:
This should be a generic class

Reply:
Generic how? Not sure what you mean sorry

Reply:
```ts
class ExampleClass<GenericTypeParameter> {
  // ...
}
```

Reply:
Okay, I know what generics are in TypeScript, but I couldn't see them being used anywhere similarly in this repository. Didn't seem to fit convention.